kolena/_experimental/classification/utils.py

```diff
@@ -251,3 +254,75 @@ class is `None`.
             curves=curves,
         )
     return None
+
+
+def compute_threshold_curves(
+    ground_truths: List[Optional[ClassificationLabel]],
+    inferences: List[ScoredClassificationLabel],
+    thresholds: Optional[List[float]] = None,
+) -> Optional[List[CurvePlot]]:
+    """
+    Computes scores (i.e. [`Precision`][kolena.workflow.metrics.precision], [`Recall`](kolena.workflow.metrics.recall)
+    and [`F1-score`](kolena.workflow.metrics.f1_score)) vs. threshold curves for a **single** class presented in
+    `inferences`.
+
+    :param ground_truths: The list of ground truth
+        [`ClassificationLabel`][kolena.workflow.annotation.ClassificationLabel]. For binary classification, the negative
+        class can be `None`.
+    :param inferences: The list of inference
+        [`ScoredClassificationLabel`][kolena.workflow.annotation.ScoredClassificationLabel]. The list should only
+         include inferences of a specific class to plot the threshold curves for.
+    :param thresholds: The list of thresholds to plot with. If not specified, all the unique confidence scores are used
+        as thresholds, including evenly spaced thresholds from 0 to 1 with 0.1 step.
+    :return: A list of [`CurvePlot`][kolena.workflow.plot.CurvePlot]s if there is any valid `Curve` computed;
+        otherwise, `None`.
+
+    """
+    if len(ground_truths) != len(inferences):
+        log.warn(f"ground_truths ({len(ground_truths)}) and inferences ({len(inferences)}) differ in length")
+        return None
+
+    if len(ground_truths) == 0 or len(inferences) == 0:
+        log.warn(
+            "insufficient # of samples to compute threshold curves — need at least 1 but received "
+            f"{len(ground_truths)}",
+        )
+        return None
+
+    inference_label = {inf.label for inf in inferences}
+    if len(inference_label) > 1:
+        log.warn(
+            f"more than one class passed in as inferences: {inference_label} — expecting inferences belonging to "
+            "a single class",
+        )
+        return None
+
+    inference_label = next(iter(inference_label))
```

Comment:
nit: not a fan of this type change for the variable -- makes things confusing to read